File: day2.py
total = 0
conversion = [{'X': 3, 'Y': 1, 'Z': 2}, {'X': 1, 'Y': 2, 'Z': 3}, {'X': 2, 'Y': 3, 'Z': 1}]

the_score_for_when_you_win = {'X': 0, 'Y':3, 'Z':6}

with open("day2input.txt") as f:
	lines = f.readlines()
	for line in lines:
		if line[0] == 'A':
			total += conversion[0][line[2]] + the_score_for_when_you_win[line[2]]
		elif line[0] == 'B':
			total += conversion[1][line[2]] + the_score_for_when_you_win[line[2]]
		elif line[0] == 'C':
			total += conversion[2][line[2]] + the_score_for_when_you_win[line[2]]
print(total)

File: aoc/src/bin/day2.rs
use std::io::{self, BufRead};
use std::fs::File;


fn main()  -> io::Result<()>{
    let file = File::open("/Users/priyashah/Documents/Personal Programming Project/Advent-Of-Code/aoc/src/files/input2.txt")?;
    let reader = io::BufReader::new(file);

    let mut safe_total = 0;

    for line in reader.lines() {
        let current_line = line.unwrap();
        let iter = current_line.split_whitespace();

        let mut mistakes: i32 = 0;
        let mut tuple = (-1, "BASE");
        let mut safe = true;

        for value in iter {
            let item = value.parse::<i32>().unwrap();
            if tuple.1 == "BASE" && tuple.0 == -1 {
                tuple.0 = item;
                continue;
            } else {
                if (tuple.0 - item).abs() > 3 {
                    if mistakes >= 1 {
                        safe = false;
                        break;
                    }
                    mistakes += 1;
                    continue;
                }
                if tuple.0 == item {
                    if mistakes >= 1 {
                        safe = false;
                        break;
                    }
                    mistakes += 1;
                    continue;
                }
                if tuple.0 > item {
                    if tuple.1 == "BASE" {
                        tuple.1 = "DECREASE";
                    }
                    else if tuple.1 == "INCREASE" {
                        if mistakes >= 1 {
                            safe = false;
                            break;
                        }
                        mistakes += 1;
                        continue;
                    }
                } 
                if tuple.0 < item {
                    if tuple.1 == "BASE" {
                        tuple.1 = "INCREASE";
                    }
                    else if tuple.1 == "DECREASE" {
                        if mistakes >= 1 {
                            safe = false;
                            break;
                        }
                        mistakes += 1;
                        continue;
                    }
                }
                tuple.0 = item;
            }
        }
        if safe == true {
            safe_total += 1;
        }
    }
    println!("Safe: {safe_total}");

    Ok(())
}